Refactor rabbitmq-script-wrapper

gerhard opened the pull request:
The logic is a bit crazy, and the whole purpose of this script is
questionable, but until we get rid of it, we've refactored to make it
more obvious just how crazy it actually is.

[#149425921]

Signed-off-by: Gerhard Lazu <gerhard@rabbitmq.com>

dumbbell committed:
Refactor rabbitmq-script-wrapper

The logic is a bit crazy, and the whole purpose of this script is
questionable, but until we get rid of it, we've refactor it to make it
more obvious just how crazy it actually is.

[#149425921]

Signed-off-by: Gerhard Lazu <gerhard@rabbitmq.com>

lukebakken reviewed (approved):
Much easier to follow. I'm assuming the previous `cd /var/lib/rabbitmq` is no longer necessary as we set `HOME` using `"$(cd ~rabbitmq && pwd)"`

dumbbell committed:
Ensure we are in a readable dir when running RabbitMQ scripts

If the current working directory is not readable, Erlang will crash.
This ensures that we run RabbitMQ scripts from a readable directory.

[#149425921]

Signed-off-by: Gerhard Lazu <gerhard@rabbitmq.com>

gerhard commented:
@lukebakken we thought so too, but learned that it's not the case across all distros: https://github.com/rabbitmq/rabbitmq-server-release/pull/35/commits/9de5b86e003d2b3589c1cea2e25fef41c3722cb1
